pkg/scheduler/cache/cache.go

```diff
@@ -294,37 +295,36 @@ func (cache *schedulerCache) processPodResourcesResizeRequest(newPod *v1.Pod) er
 			allocatable := node.AllocatableResource()
 			nodeMilliCPU := node.RequestedResource().MilliCPU
 			nodeMemory := node.RequestedResource().Memory
-
 			if (allocatable.MilliCPU > (podResource.MilliCPU + nodeMilliCPU)) &&
 				(allocatable.Memory > (podResource.Memory + nodeMemory)) {
 				// InPlace resizing is possible
 				for i, container := range newPod.Spec.Containers {
 					resizeContainer, ok := resizeContainersMap[container.Name]
 					if ok {
 						// Validation checks ensure pod QoS invariance, just update changed values
-						if resizeContainer.Resources.Requests != nil {
+						if (resizeContainer.Resources.Requests != nil) {
 							for k, v := range resizeContainer.Resources.Requests {
 								newPod.Spec.Containers[i].Resources.Requests[k] = v
 							}
 						}
-						if resizeContainer.Resources.Limits != nil {
+						if (resizeContainer.Resources.Limits != nil) {
 							for k, v := range resizeContainer.Resources.Limits {
 								newPod.Spec.Containers[i].Resources.Limits[k] = v
 							}
 						}
 					}
 				}
-				newPod.ObjectMeta.Annotations[api.AnnotationResizeResources] = api.ResizeActionUpdate
+				newPod.ObjectMeta.Annotations[api.AnnotationResizeResourcesAction] = string(api.ResizeActionUpdate)
```

Comment:
not sure why string() is needed here

Reply:
by default annotations on the pod are string type. i changed the annotations to its own types